utils.sh

```diff
@@ -73,6 +73,11 @@ function create_cluster() {
       jq -s '.[0] * .[1]' ${IGNITION_EXTRA} ${assets_dir}/worker.ign.orig | tee ${assets_dir}/worker.ign
     fi
     $OPENSHIFT_INSTALLER --dir "${assets_dir}" --log-level=debug create cluster
+
+    OCP_VERSIONS_NOAUTH="4.3 4.4 4.5"
```

Comment:
We're checking versions here and on line 246. Could we move all of this into generate_auth_template so that it does the right thing regardless of the version?

Reply:
the templates are different for old versions (I think 4.3 4.4 and 4.5 doesn't support http_basic) so we just generate the normal template on generate_templates, I also added the new template so it will generate the one to use the http_basic and it will be only call if the versions is not  in the OCP_VERSIONS_NOAUTH

Reply:
Sure, the templates are different. But we're handling templates for those different versions in different places.

Reply:
> the templates are different for old versions

That seems like something that could be templatable :)

Reply:
it's running in different places because for noauth we don't need oc cli available to get the username password, maybe it would be fine to generate everything in the end...

Reply:
Yes, I think keeping all of it together in one place will make it easier to follow and maintain later.